client/signup/steps/domains/index.jsx

```diff
@@ -656,7 +656,11 @@ class DomainsStep extends Component {
 		}
 
 		if ( isReskinned ) {
-			return ! stepSectionName && translate( 'Enter some descriptive keywords to get started' );
+			return (
+				! stepSectionName &&
+				'domain-transfer' !== this.props.flowName &&
+				translate( 'Enter some descriptive keywords to get started' )
+			);
```

Comment:
Nit. Maybe  we can be a bit more explicit here.
```suggestion
			if( stepSectionName || 'domain-transfer' === this.props.flowName ){
			    return null;
			}
			return translate( 'Enter some descriptive keywords to get started' );
```